EE.C

```diff
@@ -0,0 +1,478 @@
+  
+   TH1F *eleIso_DY = new TH1F("eleIso_DY","electron isolation",100,0,400);
+
+   TH1F *jetpT_DY = new TH1F("jetpT_DY","jet pT",100,0,500);
+   TH1F *jeteta_DY = new TH1F("jeteta_DY","jet eta",20,-5,5);
+   
+   TH1F *bjetpT_DY = new TH1F("bjetpT_DY","b-jet pT",100,0,500);
+   TH1F *bjeteta_DY = new TH1F("bjeteta_DY","b-jet eta",20,-5,5);
+
+   TH1F *LeadJetpT_DY = new TH1F("LeadJetpT_DY","Leading b-jet pT",50,0,500);
+   TH1F *SecJetpT_DY = new TH1F("SecJetpT_DY","2nd leading b-jet pT",50,0,500);
+   
+//   TH1F *btagEvent_DY = new TH1F("btagEvent_DY","btagEvent",100,0,500);
+
+   TH1F *njet_DY = new TH1F("njet_DY","Number of jet",11,0,11);
+   TH1F *nbjet_DY = new TH1F("nbjet_DY","Number of b-jet",11,0,11);
+   TH1F *id_DY = new TH1F("id_DY","particle id",30,-15,15);
+   
+   TH1F *MET_DY = new TH1F("MET_DY","Missing ET",100,0,1000);
+   TH1F *dilep_mass_DY = new TH1F("dilep_mass_DY","dilepton mass",100,0,1000);
+   
+   TH1F *SBM_DY = new TH1F("SBM_DY","max(SBM)",100,0,300);
+   TH1F *METvsMmm_DY = new TH1F("METvsMmm_DY","E^{miss}_{T}/M(#mu^{+}#mu^{-})",100,0,1);
+   TH1F *HTLT_DY = new TH1F("HTLT_DY","HT-LT",100,-500,500);
+
+   TH1F *Mass_DY = new TH1F("Mass_DY","Lepton mass(M_{e})",100,0,500);
+   
+   TH2F *njvsnbj_DY = new TH2F("njvsnbj_DY","njet vs nbjet",10,0,10,5,0,5);
+//   TH2F *bjetpT_dR_DY = new TH2F("bjetpT_dR_DY","b-jet pT vs DeltaR",100,0,500,50,0,5);
+//   TH2F *bjetpT_elepT_DY = new TH2F("bjetpT_elepT_DY","b-jet pT vs electron pT",100,0,500,100,0,500);
+//   TH2F *bjetpT_mupT_DY = new TH2F("bjetpT_mupT_DY","b-jet pT vs muon pT",100,0,500,100,0,500);
+   
+   Long64_t nentries = fChain->GetEntriesFast();
+
+   float totalN = 122055388;
+   float Lumi = 36.3;
+   float cs = 5765.4;
+   float weight = Lumi * cs/totalN;
+
+   Long64_t nbytes = 0, nb = 0;
+   for (Long64_t jentry=0; jentry<nentries;jentry++) {
+      Long64_t ientry = LoadTree(jentry);
+      if (ientry < 0) break;
+      nb = fChain->GetEntry(jentry);   nbytes += nb;
+      // if (Cut(ientry) < 0) continue;
+
+      run_DY->Fill(run);
+
+      //MET_DY->Fill(met_pt);
+      //dilep_mass_DY->Fill(dil_mass);
+
+      if(lep_pt[0] > 53 &&lep_pt[1] > 53){
+      if(dil_dR > 0.1){
+      if(fabs(lep_dz[0])<0.5 && fabs(lep_dz[1])<0.5){
+
+	  
+	 
+	  //Opposite sign
+	  if(lep_id[0] * lep_id[1] < 0){
+	 
+      //if(dil_chosen != 0) continue;
+	      //particle id
+	      int id1 = lep_id[0];
+	      int id2 = lep_id[1];
+
+	      id_DY->Fill(id1);
+	      id_DY->Fill(id2);
+	      
+	      float Lep_pt[2] = {-999,-999};
+	      float Lep_eta[2] = {-999,-999};
+	      float Lep_phi[2] = {-999,-999};
+	      float Lep_Iso[2] = {-999,-999};
+	      float Lep_E[2] = {-999,-999};
+
+	      float Dil_dPhi[1] = {-999};
+	      float Dil_dR[1] = {-999};
+	      float Dil_mass[1] = {-999};
+	      
+	      float MET[1] = {-999};
+	      
+	      int Njet = 0;
+	      int Nbjet = 0;
+	      int Nnon_bjet = 0;
+	      
+	      vector<float> bJet_pt; bJet_pt.clear();
```

Comment:
Same comment as for leptons applies here. Using a vector of TLorentzVectors instead will cut down the length of this code by a factor of 4 without using functionality.